### PR TITLE
Verify region in BIR

### PR DESCRIPTION
**Pull request by ushirask**
Verify regions in BIR

### CODE REVIEW

File: compiler/modules/bir/verify.bal [hunk 1]
```diff
@@ -70,12 +70,80 @@ public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode c
...
+            insn2 = getCondInsn(blocks, insn.ifFalse);
+            if insn2 is BranchInsn && insn2.dest == region.exit {
+                return;
+            }
+        }
+        return vc.invalidErr("loop region exit is invalid", pos=insn.pos);
+    }
+}
+
+function getCont(Insn insn, BasicBlock[] blocks) returns Insn? {
```
Comment (jclark) [inline compiler/modules/bir/verify.bal:132]: Don't know what `Cont` means.

File: compiler/modules/bir/verify.bal [hunk 2]
```diff
@@ -70,12 +70,80 @@ public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode c
...
+    foreach Region r in code.regions {
+        check verifyRegion(cx, r, code.blocks);
+    }
 }
 
 type IntBinaryInsn IntArithmeticBinaryInsn|IntBitwiseBinaryInsn;
 
 type Error err:Semantic|err:Internal;
 
+function verifyRegion(VerifyContext vc, Region region, BasicBlock[] blocks) returns Error? {
```
Comment (jclark) [inline compiler/modules/bir/verify.bal:82]: Write a comment describing precisely what this is verifying.

Comment (jclark) [reply compiler/modules/bir/verify.bal:82]: Why is `blocks` an argument? It's part of `vc`.

File: compiler/modules/bir/verify.bal [hunk 3]
```diff
@@ -70,12 +70,80 @@ public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode c
...
+    if insn is BranchInsn {
+        return getCondInsn(blocks, insn.dest);
+    }
+    else if insn is CondBranchInsn {
+        return insn;
+    }
+    return;
+}
+
+function getCondInsn(BasicBlock[] blocks, Label label) returns Insn {
```
Comment (jclark) [inline compiler/modules/bir/verify.bal:142]: I don't know what this does.  I suspect it should be a method on VerifyContext.

File: compiler/modules/bir/verify.bal (hunk 1 above)
Comment (jclark) [reply compiler/modules/bir/verify.bal:132]: I suspect it should be a method on VerifyContext.

**Comment by ushirask** [comment]
Moved to #971

**Pull request closed without merging**